src/main/java/eu/europeana/metis/sandbox/service/util/XsltUrlUpdateServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package eu.europeana.metis.sandbox.service.util;
+
+import eu.europeana.metis.sandbox.repository.TransformXsltRepository;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.nio.charset.StandardCharsets;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+public class XsltUrlUpdateServiceImpl implements XsltUrlUpdateService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XsltUrlUpdateServiceImpl.class);
+  private static final HttpClient httpClient = HttpClient.newBuilder().build();
+  private final TransformXsltRepository transformXsltRepository;
+
+  public XsltUrlUpdateServiceImpl(TransformXsltRepository transformXsltRepository) {
+    this.transformXsltRepository = transformXsltRepository;
+  }
+
+  @Override
+  @Transactional
+  public void updateXslt(String defaultXsltUrl) {
+
+    HttpRequest httpRequest = HttpRequest.newBuilder()
+        .GET()
+        .uri(URI.create(defaultXsltUrl))
+        .build();
+
+    try (final InputStream xsltStream = httpClient.send(httpRequest, BodyHandlers.ofInputStream())
+        .body()) {
+      String transformXslt = new String(xsltStream.readAllBytes(),
+          StandardCharsets.UTF_8);
+      var entity = transformXsltRepository.findByTransformXslt(transformXslt);
+      // if xslt not in DB, save it
+      if (entity != null && entity.isPresent()) {
```

Comment:
This says, if it exists in the db then store again what is already present in the database.
If it doesn't exist, it will never save anything.
So this won't do the update successfully for the first or any subsequent runs.


I think this can be simplified by always storing the retrieved value, no need for checkups.
And when there is an error do nothing to the db(the value should be already there from a previous run).

Reply:
You're correct. It's fixed now